cmd/metal-api/internal/service/machine-service.go

```diff
@@ -1143,6 +1147,32 @@ func createMachineAllocationSpec(ds *datastore.RethinkStore, machineRequest v1.M
 		return nil, fmt.Errorf("size:%s not found err:%w", sizeID, err)
 	}
 
+	if len(machineRequest.DNSServers) > 3 {
+		return nil, errors.New("please specify a maximum of three dns servers")
+	}
+	for _, dnsip := range machineRequest.DNSServers {
+		_, err := netip.ParseAddr(dnsip.IP)
+		if err != nil {
+			return nil, fmt.Errorf("IP: %s for DNS server not correct err: %w", dnsip, err)
+		}
+	}
+
+	if len(machineRequest.NTPServers) <= 3 || len(machineRequest.NTPServers) > 5 {
```

Comment:
Should an empty list be allowed, too? In the MEP it says that specifying the NTP servers is optional. And if not provided the distro defaults are used.

Reply:
Sure, can be totally omitted, but if given len must be between 3 and 5

@simcod this must be fixed obviously